### PR TITLE
Datagraph - 1075 - Fix Page's element count calculation

### PR DESCRIPTION
**Pull request by meistermeier**
This PR will solve a problem with calculating the right count of total elements when using PageRequests.

### CODE REVIEW

File: spring-data-neo4j/src/test/java/org/springframework/data/neo4j/repository/support/SimpleNeo4jRepositoryTests.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import org.neo4j.ogm.cypher.query.SortOrder;
+import org.neo4j.ogm.session.Session;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+
+public class SimpleNeo4jRepositoryTests {
+
+	private final Session sessionMock = mock(Session.class);
+	private final PageRequest twoElementPageRequest = PageRequest.of(0, 2);
+	private final PageRequest oneElementPageRequest = PageRequest.of(0, 1);
```
Comment (michael-simons) [inline spring-data-neo4j/src/test/java/org/springframework/data/neo4j/repository/support/SimpleNeo4jRepositoryTests.java]: I'd like to see a test in the middle of something, like 5 elements in total, requesting 2nd page with pagesize 2 as well.

File: spring-data-neo4j/src/main/java/org/springframework/data/neo4j/repository/support/SimpleNeo4jRepository.java [hunk 2]
```diff
@@ -190,24 +189,9 @@ public void deleteAll() {
...
-	private Page<T> updatePage(Pageable pageable, List<T> results) {
-
-		int pageSize = pageable.getPageSize();
-		long pageOffset = pageable.getOffset();
-		long total = pageOffset + results.size() + (results.size() == pageSize ? pageSize : 0);
+		Pagination pagination = new Pagination(pageable.getPageNumber(), pageable.getPageSize());
+		Collection<T> data = session.loadAll(clazz, PagingAndSortingUtils.convert(pageable.getSort()), pagination, depth);
 
-		return new PageImpl<T>(results, pageable, total);
+		return PageableExecutionUtils.getPage(new ArrayList<>(data), pageable, () -> session.countEntitiesOfType(clazz));
```
Comment (michael-simons) [inline spring-data-neo4j/src/main/java/org/springframework/data/neo4j/repository/support/SimpleNeo4jRepository.java]: To bad that `getPage` needs a list, not a collection. I just checked, we return a `Set` and so we are copying stuff around.

Comment (meistermeier) [reply spring-data-neo4j/src/main/java/org/springframework/data/neo4j/repository/support/SimpleNeo4jRepository.java]: There will be no API change in SD commons in the near future. As far as we checked this `Collection` is already an `ArrayList` but I won't do a downcast here.

Comment (michael-simons) [reply spring-data-neo4j/src/main/java/org/springframework/data/neo4j/repository/support/SimpleNeo4jRepository.java]: Fully understandable.

**Comment by meistermeier** [comment]
Merged with commit cf3ae693e46b7107eb535040d08427b73d6f3045.

**Pull request closed without merging**